Tasks/AzureFunctionAppV2/azure-arm-rest/azure-arm-app-service.ts

```diff
@@ -721,6 +721,31 @@ export class AzureAppService {
         return this._name;
     }
 
+    public async getSitePublishingCredentialPolicies(): Promise<any> {
+        try {
+            var httpRequest = new webClient.WebRequest();
+            httpRequest.method = 'GET';
+            var slotUrl: string = !!this._slot ? `/slots/${this._slot}` : '';
+            httpRequest.uri = this._client.getRequestUri(`//subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Web/sites/{name}/${slotUrl}/basicPublishingCredentialsPolicies/scm`,
+            {
+                '{resourceGroupName}': this._resourceGroup,
+                '{name}': this._name,
+            }, null, '2022-03-01');
+            let requestOptions = new webClient.WebRequestOptions();
+            requestOptions.retryCount = 1;
+            
+            var response = await this._client.beginRequest(httpRequest, requestOptions);
```

Comment:
>requestOptions [](http://example.com/codeflow?start=72&length=14)

In common-npm-packages\azure-arm-rest\azure-arm-app-service.ts

beginRequest just takes a single parameter -- so can't pass in retryCount (possibly due to node differences).  Will that be an issue?

Reply:
Hi. we're actually using import webClient = require('azure-pipelines-tasks-azurermdeploycommon/azure-arm-rest/webClient'); which does support Retry count I thought? https://github.com/microsoft/azure-pipelines-tasks/blob/master/common-npm-packages/azure-arm-rest-v2/webClient.ts#L41

Reply:
> webClient = require('azure-pipelines-tasks-azurermdeploycommon/azure-arm-rest/webClient'); which does support Retry count I thought?

I see and I don't have any objection to having the retry here.

However, in common rm library, unless it's necessary, I'd prefer not to change to the wrapper to minimize the churn as much as possible.   Is the retry important?

Reply:
I think this is important as it mirrors the other calls to site checks we do and is the general guidance for Azure to have some retry?